spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/query/PartTreeDatastoreQueryTests.java

```diff
@@ -809,263 +798,252 @@ private void prepareDeleteResults(boolean isCollection) {
+                            + "ThanEqualAndIdIsNullOrderByIdDesc");
     this.partTreeDatastoreQuery = createQuery(false, false, null);
 
     // There are too few params specified, so the exception will occur.
     Object[] params = new Object[] {"BUY"};
 
-    this.partTreeDatastoreQuery.execute(params);
+    assertThatThrownBy(() -> this.partTreeDatastoreQuery.execute(params))
+            .hasMessage("Too few parameters are provided for query method: "
+                    + "findByActionAndSymbolAndPriceLessThanAndPriceGreaterThanEqualAndIdIsNullOrderByIdDesc");
   }
 
   @Test
-  public void unsupportedParamTypeTest() throws NoSuchMethodException {
-    this.expectedException.expectMessage(
-        "Unable to convert class "
-            + "com.google.cloud.spring.data.datastore.repository.query."
-            + "PartTreeDatastoreQueryTests$Trade to Datastore supported type.");
+  void unsupportedParamTypeTest() throws NoSuchMethodException {
+
     queryWithMockResult("findByAction", null, getClass().getMethod("countByPrice", Integer.class));
 
     this.partTreeDatastoreQuery = createQuery(false, false, null);
 
     Object[] params = new Object[] {new Trade()};
 
-    this.partTreeDatastoreQuery.execute(params);
+    assertThatThrownBy(() ->  this.partTreeDatastoreQuery.execute(params))
+            .hasMessage("Unable to convert class "
+                    + "com.google.cloud.spring.data.datastore.repository.query."
+                    + "PartTreeDatastoreQueryTests$Trade to Datastore supported type.");
   }
 
   @Test
-  public void unSupportedPredicateTest() throws NoSuchMethodException {
-    this.expectedException.expectMessage("Unsupported predicate keyword: BETWEEN");
+  void unSupportedPredicateTest() throws NoSuchMethodException {
 
     queryWithMockResult("countByTraderIdBetween", null, getClass().getMethod("traderAndPrice"));
     this.partTreeDatastoreQuery = createQuery(false, false, null);
-    this.partTreeDatastoreQuery.execute(EMPTY_PARAMETERS);
+
+    assertThatThrownBy(() -> this.partTreeDatastoreQuery.execute(EMPTY_PARAMETERS))
+            .hasMessageContaining("Unsupported predicate keyword: BETWEEN");
+
+
   }
 
   @Test
-  public void unSupportedOrTest() throws NoSuchMethodException {
-    this.expectedException.expectMessage(
-        "Cloud Datastore only supports multiple filters combined with AND");
+  void unSupportedOrTest() {
+
+    assertThatThrownBy(() -> queryWithMockResult("countByTraderIdOrPrice", null, getClass().getMethod("traderAndPrice")))
+            .hasMessage("Cloud Datastore only supports multiple filters combined with AND.");
 
-    queryWithMockResult("countByTraderIdOrPrice", null, getClass().getMethod("traderAndPrice"));
 
     // this.partTreeDatastoreQuery = createQuery();
-    this.partTreeDatastoreQuery.execute(new Object[] {123L, 45L});
+    //this.partTreeDatastoreQuery.execute(new Object[] {123L, 45L});
```

Comment:
This test was unfortunately structured in the first place with mock setup faling. Could you remove the commented-out lines, and add a comment to this test saying that "PartTreeDatastoreQuery constructor will fail as part of queryWithMockResult setup"?

The commented out line that was there in the first place shows why it was done this way -- getting all the setup in place before calling the constructor was too hard. This is commonly a sign that code under test has a structure that does not lend itself to testing.